src/ds/features/auto-exposure-gain-limit-feature.cpp

```diff
@@ -0,0 +1,68 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+
+#include <src/ds/features/auto-exposure-gain-limit-feature.h>
+#include <src/ds/d400/d400-options.h>
+#include <src/ds/ds-private.h>
+#include <src/sensor.h>
+
+
+namespace librealsense {
+
+const feature_id auto_exposure_gain_limit_feature::ID = "Auto Exposure Gain Limit feature";
+
+auto_exposure_gain_limit_feature::auto_exposure_gain_limit_feature( synthetic_sensor & sensor, std::shared_ptr< hw_monitor > hw_monitor )
+{
+    auto exposure_range = sensor.get_option( RS2_OPTION_EXPOSURE ).get_range();
+    auto gain_range = sensor.get_option( RS2_OPTION_GAIN ).get_range();
+
+    bool auto_exposure_new_opcode = false;
```

Comment:
Why initialize a value you are about to change? Declare and initialize in the same expression.